src/tools/osd-target-run/osd-target-run.c

```diff
@@ -369,6 +368,82 @@ static osd_result run_tracing(void)
     return retval;
 }
 
+static osd_result run_terminal(void)
+{
+    struct osd_module_desc *modules;
+    struct osd_hostmod_ctx *hostmod_enum;
+    osd_result rv;
+    size_t modules_len;
+
+    // We only create the pseudo-terminal if it was explicitly specified
+    if (!a_terminal->count) {
+        return OSD_OK;
+    }
+
+    rv = osd_hostmod_new(&hostmod_enum, osd_log_ctx, HOSTCTRL_EP, NULL, NULL);
+    if (OSD_FAILED(rv)) {
+        goto free_return;
+    }
+
+    rv = osd_hostmod_connect(hostmod_enum);
+    if (OSD_FAILED(rv)) {
+        goto free_return;
+    }
+
+    rv = osd_hostmod_get_modules(hostmod_enum, DEVICE_SUBNET_ADDRESS, &modules,
+                                 &modules_len);
+    if (OSD_FAILED(rv)) {
+        goto free_return;
+    }
+
+    for (size_t i = 0; i < modules_len; i++) {
+        if (modules[i].vendor == OSD_MODULE_VENDOR_OSD &&
+            modules[i].type == OSD_MODULE_TYPE_STD_DEM_UART) {
+            rv = osd_terminal_new(&terminal_ctx, osd_log_ctx, HOSTCTRL_EP,
+                                  modules[i].addr);
+            if (OSD_FAILED(rv)) {
+                fatal("osd_terminal_new() failed with code: %i\n", rv);
```

Comment:
"\n" is added automatically to the fatal() output, isn't it? (Same also for the other places in the code.)